pkg/subscriptions/subscriptions.js

```diff
@@ -18,594 +18,104 @@
-
-        /* The promise for the last operation */
-        var last_promise;
-
-        function status_update_failed() {
-            if (update_timeout !== null)
-                $('#subscription-manager-not-found').show();
-            update_timeout = null;
-        }
-
-        function get_overall_status() {
-            return overall_status;
-        }
-
-        function render_subscriptions(info) {
-            /* clear old subscription details */
-            subscription_info.empty();
-
-            /* show subscription info only if there is visible information */
-            if (info.length > 0) {
-                subscription_info.show();
-                /* render and add the new ones */
-                for (var i = 0; i < info.length; ++i)
-                    subscription_info.append( $(Mustache.render(subscription_template, info[i])) );
+    "subscriptions/react",
+    "subscriptions/subscriptions-client",
+    "subscriptions/subscriptions-register",
+    "subscriptions/subscriptions-view",
+    "subscriptions/cockpit-components-dialog",
+], function(cockpit, React, subscriptionsClient, subscriptionsRegister, subscriptionsView, Dialog) {
+
+"use strict";
+
+var _ = cockpit.gettext;
+
+var dataStore = { };
+
+function dataChanged() {
+    dataStore.render();
+}
+
+function dismissStatusError() {
+    subscriptionsClient.subscriptionStatus.error = undefined;
+    dataStore.render();
+}
+
+var registerDialogDetails;
+
+function registerSystem () {
+    return subscriptionsClient.registerSystem(registerDialogDetails);
+}
+
+var footerProps = {
+    'actions': [
+          { 'clicked': registerSystem,
+            'caption': _("Register"),
+            'style': 'primary',
+          },
+      ]
+};
+
+function openRegisterDialog() {
+    registerDialogDetails = subscriptionsRegister.defaultSettings();
+
+    // show dialog to register
+    var renderDialog;
+    var updatedData = function(prop, data) {
+        if (prop) {
+            if (data.target) {
+                if (data.target.type == "checkbox") {
+                    registerDialogDetails[prop] = data.target.checked;
+                } else {
+                    registerDialogDetails[prop] = data.target.value;
+                    // input from the ui, so we don't need to re-render
+                    return;
+                }
             } else {
-                subscription_info.hide();
+                registerDialogDetails[prop] = data;
             }
         }
 
-        /*
-         * Parses lines like:
-         *
-         * id:  text
-         */
-        function parse_single_subscription(text) {
-            var ret = { };
-            $.each(text.split('\n'), function(i, line) {
-                var pos = line.indexOf(':');
-                if (pos !== -1)
-                    ret[line.substring(0, pos)] = line.substring(pos + 1).trim();
-            });
-            return ret;
-        }
+        registerDialogDetails.onChange = updatedData;
```

Comment:
Note that event target works great with things like setting ```xxx.onchange``` to a function.

So for example with:

```
registerDialogDetails.onchange = function(ev, prop, data) { console.log("yelloooo", prop, data); };
```

Will trigger with this code (obviously) elsewhere and appropriately split up:

```
cockpit.event_target(registerDialogDetails);
...
registerDialogDetails.dispatchEvent('change', prop, data);
```